### PR TITLE
Update docs: changes to creating a release

### PR DESCRIPTION
**Pull request by brooksn**
#### Proposed Changes ####

Add missing step to RC release.

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

**Review by brandond — APPROVED** [review]

File: developer-docs/upgrading_kubernetes.md [hunk 1]
```diff
@@ -54,9 +54,10 @@ Once these changes are made, submit a PR for review and let CI complete. When CI
 
 Next, we need to create a release candidate (RC). The Drone (CI) process that builds the release itself can be monitored [here](https://drone-publish.rancher.io/rancher/rke2/).
 
-* Click "Releases"
-* Click "Draft new release"
-* Enter the desired version into the "Tag version" box. 
+1. Click "Releases"
```
Comment (briandowns) [inline developer-docs/upgrading_kubernetes.md:57]: Let's keep the asterisks here to be consistent with the reset of the docs and the ones in different branches. We can do a full reformatting in a separate PR.

**Review by jossemargt-3pillar — APPROVED** [review]
